packages/react/src/base-chart.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Chart as ChartType } from "@carbon/charts/chart";
+import { ChartTabularData, BaseChartOptions, ChartConfig } from "@carbon/charts/interfaces";
+
+type Props<Options, Data> = { options?: Options; data?: Data } | ChartConfig<Options>;
+
+export default class BaseChart<
+  Options = BaseChartOptions,
+  Chart = ChartType,
+  Data = ChartTabularData,
+> extends React.Component<Props<Options, Data>> {
+  data: Data | {};
+  options: Options | {};
+  props!: Props<Options, Data>;
+  chart!: Chart;
+
+  constructor(props: Props<Options, Data>) {
+    super(props);
+
+    const { options, data } = props;
+
+    if (!options) {
+      console.error("Missing options!");
+    }
+
+    if (!data) {
+      console.error("Missing data!");
+    }
+
+    this.data = props.data || {};
+    this.options = props.options || {};
+
+    Object.assign(this, this.chart);
+  }
+
+  shouldComponentUpdate(nextProps: Props<Options, Data>) {
+    return (
+      this.props.data !== nextProps.data ||
+      this.props.options !== nextProps.options
+    );
+  }
+
+  componentDidUpdate() {
+    // @ts-ignore
```

Comment:
curios as to why these are needed? do we need to fix something in the `setData` method? add some typing maybe?

Reply:
not necessarily requesting a change here, more so looking for the problem

Reply:
Fixed in https://github.com/carbon-design-system/carbon-charts/pull/867/commits/e9a53f4336a7af42a6b9caf555185df5624fc4de#diff-827cd8b812b357745e517c845fa7d5ca9f9d0cb61fdb0e50d285f809ff9d631b

Reply:
Nice 🎉